File: ride-management/src/main/kotlin/com/example/ridemanagement/domain/model/CompletedRide.kt
package com.example.ridemanagement.domain.model

import com.example.sharedkernel.domain.base.AbstractEntity
import com.example.sharedkernel.domain.base.DomainObjectId
import com.example.sharedkernel.domain.id.RideId
import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "completed_rides")
data class CompletedRide(
        @Embedded
        @AttributeOverrides(
                AttributeOverride(name = "id", column = Column(name = "ride_id", nullable = false))
        )
        val rideId: RideId,

        @Column(nullable = false)
        val completedOn: LocalDateTime,

        @ManyToOne
        val conductor: Conductor
) : AbstractEntity<CompletedRideId>(DomainObjectId.randomId(CompletedRideId::class.java))

File: jsp-trams-frontend/src/Models/CompletedRide.ts
import { RideId } from "Models/Ride";
import { Conductor } from "Models/Conductor";

interface CompletedRideId {
    id: string;
}

export interface CompletedRide {
    id: CompletedRideId;
    rideId: RideId;
    completedOn: string;
    conductor: Conductor;
}
